lua/entities/gmod_wire_expression2/core/entity.lua

```diff
@@ -45,6 +45,37 @@ local rad2deg = 180 / math.pi
 
 /******************************************************************************/
 
+local E2TABLE_DEFAULT = { n = {}, ntypes = {}, s = {}, stypes = {}, size = 0 }
+local LuaTypeToWireTypeID
+do
+	local type, isnumber, isstring, string_upper, wire_expression_types = type, isnumber, isstring, string.upper, wire_expression_types
+	local type_conversions = {
+		["boolean"] = function(value) return "normal", value and 1 or 0 end,
+		["Color"  ] = function(value) return "Vector4", { value.r, value.g, value.b, value.a },
+		["Angle"  ] = function(value) return "Angle", { value.p, value.y, value.r } end,
+		["Vector" ] = function(value) return "Vector", { value.x, value.y, value.z } end,
+		["NPC"    ] = function() return "Entity" end,
+		["Player" ] = function() return "Entity" end,
+		["Vehicle"] = function() return "Entity" end
+	}
+
+	LuaTypeToWireTypeID = function(value)
+		local typename = type(value)
+		if type_conversions[typename] then
+			local new_val
+			typename, new_val = type_conversions[typename](value)
+			if new_val ~= nil then
+				value = new_val
+			end
+		elseif typename == "number" then
+			typename = "normal"
+		end
+		return wire_expression_types[string_upper(typename)][1], value
+	end
+end
+
+/******************************************************************************/
```

Comment:
It is consistent with the rest of the code in this file, or do you want me to change it to Lua comment instead?
You could easily do this small change yourself, that's why I have allowed edits from maintainers :)